libs/langchain/langchain/llms/google_palm.py

```diff
@@ -117,31 +173,164 @@ def _generate(
+                        run_manager=run_manager,
+                        generation_config=generation_config,
+                    )
+                    candidates = [
+                        "".join([p.text for p in c.content.parts])
+                        for c in res.candidates
+                    ]
+                    generations.append([Generation(text=c) for c in candidates])
+                else:
+                    res = completion_with_retry(
+                        self,
+                        model=self.model_name,
+                        prompt=prompt,
+                        stream=should_stream,
+                        is_gemini=False,
+                        run_manager=run_manager,
+                        **generation_config,
+                    )
+                    prompt_generations = []
+                    for candidate in res.candidates:
+                        raw_text = candidate["output"]
+                        stripped_text = _strip_erroneous_leading_spaces(raw_text)
+                        prompt_generations.append(Generation(text=stripped_text))
+                    generations.append(prompt_generations)
+
+        return LLMResult(generations=generations)
+
+    def _stream(
+        self,
+        prompt: str,
+        stop: Optional[List[str]] = None,
+        run_manager: Optional[CallbackManagerForLLMRun] = None,
+        **kwargs: Any,
+    ) -> Iterator[GenerationChunk]:
+        generation_config = kwargs.get("generation_config", {})
+        if stop:
+            generation_config["stop_sequences"] = stop
+        for stream_resp in completion_with_retry(
+            self,
+            prompt,
+            stream=True,
+            is_gemini=True,
+            run_manager=run_manager,
+            generation_config=generation_config,
+            **kwargs,
+        ):
+            chunk = GenerationChunk(text=stream_resp.text)
+            yield chunk
+            if run_manager:
+                run_manager.on_llm_new_token(
+                    stream_resp.text,
+                    chunk=chunk,
+                    verbose=self.verbose,
+                )
+
+    def invoke(
+        self,
+        input: LanguageModelInput,
+        config: Optional[RunnableConfig] = None,
+        *,
+        stop: Optional[List[str]] = None,
+        **kwargs: Any,
+    ) -> str:
+        generation_config = {
+            "stop_sequences": stop,
+            "temperature": self.temperature,
+            "top_p": self.top_p,
+            "top_k": self.top_k,
+            "max_output_tokens": self.max_output_tokens,
+            "candidate_count": self.n,
+        }
+        if isinstance(input, str):
+            return super().invoke(input, config, stop=stop, **generation_config)
+        elif isinstance(input, StringPromptValue):
+            return super().invoke(
+                input.to_string(), config, stop=stop, **generation_config
+            )
+        elif isinstance(input, list):
```

Comment:
This logic isn't correct -> vision inputs would be passed as a list within the `content`. As it stands, if you did
```
messages = [HumanMessage(content="HI there"), AIMessage(content="How can I help"), HumanMessage(content="What's the time?")]
GoogleGenerativeAi(model_name="gemini-pro").invoke(messages)
````

it would erroneously raise an error

Reply:
maybe I misunderstand the the `invoke` method then. You should pass >1 messages only to the ChatModel, or am I wrong?

And from what I see here, vision input is passed as part of the same message (a `content` is a list, but the model gets a list with a single message): 
https://github.com/langchain-ai/langchain/blob/c0f4b95aa9961724ab4569049b4c3bc12ebbacfc/templates/rag-chroma-multi-modal/rag_chroma_multi_modal/chain.py#L74

https://github.com/langchain-ai/langchain/blob/master/cookbook/Multi_modal_RAG.ipynb

Reply:
Removed multimodality, it'll be implemented on a corresponding chat model.